### PR TITLE
zh-CN: Update the translation of `right shift`

### PR DESCRIPTION
**Pull request by RC-Kanashii**
### Description

Update the translation of `right shift`.

### Motivation

The simplified Chinese translation version of `right shift` is out of date.

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/operators/right_shift/index.md [hunk 1]
```diff
@@ -1,51 +1,83 @@
 ---
-title: 右移 (>>)
+title: 右移（>>）
 slug: Web/JavaScript/Reference/Operators/Right_shift
 ---
 
 {{jsSidebar("Operators")}}
 
-**右移操作符 (`>>`)** 是将一个操作数按指定移动的位数向右移动，右边移出位被丢弃，左边移出的空位补符号位（最左边那位）。
+**右移运算符（`>>`）**将一个操作数的二进制表示形式向右移动指定位数，该操作数可以是 `Number` 类型或者 `BigInt` 类型。右边移出位被丢弃，左边移出的空位补符号位（最左边那位）。该操作也称为“符号位传播右移”（sign-propagating right shift）或“算术右移”（arithmetic right shift），因为返回值的符号位与第一个操作数的符号位相同。
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/operators/right_shift/index.md:8]:
```suggestion
**右移运算符（`>>`）** 将一个操作数的二进制表示形式向右移动指定位数，该操作数可以是 `Number` 类型或者 `BigInt` 类型。右边移出位被丢弃，左边移出的空位补符号位（最左边那位）。该操作也称为“符号位传播右移”（sign-propagating right shift）或“算术右移”（arithmetic right shift），因为返回值的符号位与第一个操作数的符号位相同。
```

File: files/zh-cn/web/javascript/reference/operators/right_shift/index.md [hunk 2]
```diff
@@ -1,51 +1,83 @@
...
-a >> b
+x >> y
 ```
 
 ## 描述
 
-**右移操作符 (`>>`)** 是将一个操作数按指定移动的位数向右移动。
-右边移出位将被丢弃，然后用最左边的这一位（符号位）填充左边的空位。
-由于新的数字最左边位与之前数字的最左边位是相同值，故符号位（最左边的位）不会改变，因此被称为“符号位传播”（sign-propagating）.
+**右移运算符（`>>`）**针对 `Number` 和 [`BigInt`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/BigInt) 两种类型的操作数进行了重载。对于 `Number` 类型，该运算符返回一个32位整数；对于 `BigInt` 类型，该运算符也返回一个 `BigInt` 。右移运算符首先[将两个操作数强制转换为数值](/zh-CN/docs/Web/JavaScript/Data_structures#%E5%BC%BA%E5%88%B6%E6%95%B0%E5%AD%97%E7%B1%BB%E5%9E%8B%E8%BD%AC%E6%8D%A2)并测试它们的类型。如果两个操作数都转换成 `BigInt`，则执行 `BigInt` 右移； 否则，它将两个操作数都转换为 [32 位整数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number#%E5%9B%BA%E5%AE%9A%E5%AE%BD%E5%BA%A6%E6%95%B0%E5%80%BC%E8%BD%AC%E6%8D%A2)并执行 `Number` 类型右移。 如果一个操作数变为 `BigInt` 而另一个变为 `Number`，则会抛出 {{jsxref("TypeError")}}。
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/operators/right_shift/index.md:20]:
```suggestion
**右移运算符（`>>`）** 针对 `Number` 和 [`BigInt`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/BigInt) 两种类型的操作数进行了重载。对于 `Number` 类型，该运算符返回一个32位整数；对于 `BigInt` 类型，该运算符也返回一个 `BigInt` 。右移运算符首先[将两个操作数强制转换为数值](/zh-CN/docs/Web/JavaScript/Data_structures#%E5%BC%BA%E5%88%B6%E6%95%B0%E5%AD%97%E7%B1%BB%E5%9E%8B%E8%BD%AC%E6%8D%A2)并测试它们的类型。如果两个操作数都转换成 `BigInt`，则执行 `BigInt` 右移； 否则，它将两个操作数都转换为 [32 位整数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number#%E5%9B%BA%E5%AE%9A%E5%AE%BD%E5%BA%A6%E6%95%B0%E5%80%BC%E8%BD%AC%E6%8D%A2)并执行 `Number` 类型右移。 如果一个操作数变为 `BigInt` 而另一个变为 `Number`，则会抛出 {{jsxref("TypeError")}}。
```

Comment (RC-Kanashii) [reply files/zh-cn/web/javascript/reference/operators/right_shift/index.md]: 已解决

File: files/zh-cn/web/javascript/reference/operators/right_shift/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (RC-Kanashii) [reply files/zh-cn/web/javascript/reference/operators/right_shift/index.md]: 已解决

File: files/zh-cn/web/javascript/reference/operators/right_shift/index.md [hunk 3]
```diff
@@ -1,51 +1,83 @@
...
 
--9 >> 2 (base 10): 11111111111111111111111111111101 (base 2) = -3 (base 10)
+```js
+丢弃前: 11100110111110100000000000000110000000000001
+丢弃后:             10100000000000000110000000000001
 ```
 
+右操作数将被转换为无符号 32 位整数，然后取模 32，因此实际移位偏移量将始终是 0 到 31 之间的正整数，包括 0 和 31。 例如，`100 >> 32` 与 `100 >> 0` 相同（结果都是 `100`），因为 32 模 32 为 0。
+
+`BigInt` 在使用右移时不会产生截断现象。 从概念上讲，正 `BigInt` 具有无限数量的前导位 `0`，而负 `BigInt` 具有无限数量的前导位 `1`。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/operators/right_shift/index.md:68]:
```suggestion
BigInt 在使用右移时不会产生截断现象。 从概念上讲，正 BigInt 具有无限数量的前导位 `0`，而负 BigInt 具有无限数量的前导位 `1`。
```

Comment (RC-Kanashii) [reply files/zh-cn/web/javascript/reference/operators/right_shift/index.md]: 已修改

**Review by yin1999 — APPROVED** [review]
Thanks. Welcome to the community :tada:

**Pull request merged**